### ISSUE TITLE
wrong decimal year conversions

### ISSUE THREAD

**Issue by pacesm** [ESA-VirES/MagneticModel#16]
It seems that the decimal year conversion used by the magnetic model (and VirES in general) is wrong.

VirES converts the decimal year so that the whole year value is always aligned with 00:00Z, 1st of January. This seems to be incorrect [[scientific decimal time](https://en.wikipedia.org/wiki/Decimal_time#Scientific_decimal_time)] and the conversion should be made as
```
MJD2000 = (decimal_year - 2000) * 365.25
decimal_year = 2000 + MJD2000 / 365.25
```

The small time offset in the decimal year time causes deviation of the calculated model values when compared with the reference implementation. E.g., for CHAOS-6-Core (x7):

![image](https://user-images.githubusercontent.com/4050019/48213472-2de27380-e37e-11e8-87f0-d2a80227bba5.png)

After correcting the time conversions this deviation drops to zero:
![image](https://user-images.githubusercontent.com/4050019/48214571-86b30b80-e380-11e8-8c82-e3b6f48ce9c6.png)

**Comment by pacesm** [comment]
After some digging, I found that the decimal year conversion and coefficients interpolation differs from one model implementation to other.

Therefore, instead of hard-coding one method for all I made the decimal year conversion and time interpolation method configurable.

By default the SHC models:
- use decimal years aligned with the calendar years
- and interpolate coefficients in the MJD2000 time domain

as agreed with NiO.

But, in addition, the SHC model loader allows optional custom decimal year conversion and/or switching to interpolation in the decimal year domain.

This will allow to tune individual models to be identical with their reference implementations.
